frontend/datagathering.cpp

```diff
@@ -96,23 +104,26 @@ void MainWindow::handle_result_cycles(HttpRequestWorker *worker)
             viewMenu_->addAction(checkableAction);
             connect(checkBox, SIGNAL(stateChanged(int)), this, SLOT(columnHider(int)));
 
-            desiredHeader_ = getFields(ui_->instrumentsBox->currentText(), ui_->instrumentsBox->currentData().toString());
-
             // Filter table based on desired headers
-            if (desiredHeader_.contains(key))
+            auto it = std::find_if(desiredHeader_.begin(), desiredHeader_.end(),
+                                   [key](const std::pair<QString, QString> &data) { return data.first == key; });
```

Comment:
Can use `auto` here to make it even neater:
```suggestion
                                   [key](const auto &data) { return data.first == key; });
```